protocols/noise/src/io/handshake.rs

```diff
@@ -297,121 +307,74 @@ impl<T> State<T>
 //////////////////////////////////////////////////////////////////////////////
 // Handshake Message Futures
 
-// RecvEmpty -----------------------------------------------------------------
-
 /// A future for receiving a Noise handshake message with an empty payload.
-///
-/// Obtained from [`Handshake::recv_empty`].
 async fn recv_empty<T>(state: &mut State<T>) -> Result<(), NoiseError>
 where
-    T: AsyncRead
+    T: AsyncReadExt + Unpin
```

Comment:
Should be `AsyncRead`, not `AsyncReadExt`. `AsyncReadExt` is automatically implemented on anything that is `AsyncRead`. Same everywhere else you do that.